### PR TITLE
2576 ethrpc audit cleanup

### PR DESCRIPTION
**Pull request by tinybike**
Tagging @adrake33 for a consistency check across many repos :grin:

### CODE REVIEW

File: src/modules/auth/actions/update-assets.js [hunk 1]
```diff
@@ -27,8 +27,9 @@ export function updateAssets(callback = logError) {
         if (allAssetsLoaded(balances)) callback(null, balances)
       })
     })
-    augur.rpc.eth.getBalance([loginAccount.address, 'latest'], (attoEthBalance) => {
-      if (!attoEthBalance || attoEthBalance.error) return callback(attoEthBalance)
+    augur.rpc.eth.getBalance([loginAccount.address, 'latest'], (err, attoEthBalance) => {
+      if (err) return callback(err)
+      if (attoEthBalance == null) return callback(attoEthBalance)
```
Comment (justinbarry) [inline src/modules/auth/actions/update-assets.js]: Missing test coverage of the 'attoEthBalance' callback.

File: src/modules/auth/actions/use-unlocked-account.js [hunk 2]
```diff
@@ -12,8 +12,9 @@ export const useUnlockedAccount = (unlockedAddress, callback = logError) => (dis
     dispatch(updateIsLoggedAndLoadAccountData(unlockedAddress, ACCOUNT_TYPES.META_MASK))
     return callback(null)
   }
-  augur.rpc.isUnlocked(unlockedAddress, (isUnlocked) => {
-    if (isUnlocked == null || isUnlocked.error) return callback(isUnlocked || 'error calling augur.rpc.isUnlocked')
+  augur.rpc.isUnlocked(unlockedAddress, (err, isUnlocked) => {
+    if (err) return callback(err)
+    if (isUnlocked == null) return callback('error calling augur.rpc.isUnlocked')
```
Comment (justinbarry) [inline src/modules/auth/actions/use-unlocked-account.js]: Missing test coverage of null 'isUnlocked' error.

File: src/modules/portfolio/actions/collect-market-creator-fees.js [hunk 3]
```diff
@@ -9,7 +9,8 @@ export const collectMarketCreatorFees = (marketId, callback = logError) => (disp
     augur.api.Cash.getBalance({ _address: marketMailboxAddress }, (err, cashBal) => {
       if (err) return callback(err)
       const cashBalance = speedomatic.bignum(cashBal)
-      augur.rpc.eth.getBalance([marketMailboxAddress, 'latest'], (attoEthBalance) => {
+      augur.rpc.eth.getBalance([marketMailboxAddress, 'latest'], (err, attoEthBalance) => {
+        if (err) return callback(err)
```
Comment (justinbarry) [inline src/modules/portfolio/actions/collect-market-creator-fees.js]: Missing test coverage.

File: src/modules/app/actions/init-augur.js [hunk 4]
```diff
@@ -24,7 +24,8 @@ function pollForAccount(dispatch, getState) {
   let account
 
   setInterval(() => {
-    AugurJS.augur.rpc.eth.accounts((accounts) => {
+    AugurJS.augur.rpc.eth.accounts((err, accounts) => {
+      if (err) return console.error(err)
```
Comment (justinbarry) [inline src/modules/app/actions/init-augur.js]: Missing test coverage.

File: src/modules/portfolio/actions/collect-market-creator-fees.js (hunk 3 above)
Comment (adrake33) [inline src/modules/portfolio/actions/collect-market-creator-fees.js]: Shouldn't there also be a check here for:
```
if (attoEthBalance == null) return callback(attoEthBalance)
```
like in src/modules/auth/actions/update-assets.js above?

File: test/auth/actions/use-unlocked-account-test.js [hunk 5]
```diff
@@ -33,9 +33,10 @@ describe(`modules/auth/actions/use-unlocked-account.js`, () => {
       '../helpers/is-meta-mask': IsMetaMask,
     })
     sinon.stub(AugurJS.augur.rpc, 'isUnlocked').callsFake((address, callback) => {
-      t.stub.augur.rpc.isUnlocked(address, (isUnlocked) => {
-        store.dispatch({ type: 'AUGURJS_RPC_IS_UNLOCKED', data: { isUnlocked } })
-        callback(isUnlocked)
+      t.stub.augur.rpc.isUnlocked(address, (err, isUnlocked) => {
+        store.dispatch({ type: 'AUGURJS_RPC_IS_UNLOCKED', data: { isUnlocked: err || isUnlocked } })
+        if (err) return callback(err)
```
Comment (adrake33) [inline test/auth/actions/use-unlocked-account-test.js]: Seems like this should also have a check like: 
```
if (isUnlocked == null) return callback('error calling augur.rpc.isUnlocked')
```

Comment (tinybike) [reply test/auth/actions/use-unlocked-account-test.js]: `rpc.isUnlocked` cannot output a `null`.  If it outputs `undefined`, that should come back as an error!

File: src/modules/portfolio/actions/collect-market-creator-fees.js (hunk 3 above)
Comment (tinybike) [reply src/modules/portfolio/actions/collect-market-creator-fees.js]: I don't think that `eth.getBalance` can return `null` or `undefined`.  Even for a brand new account it should just return 0.

File: src/modules/auth/actions/use-unlocked-account.js (hunk 2 above)
Comment (tinybike) [reply src/modules/auth/actions/use-unlocked-account.js]: Actually, AFAIK `rpc.isUnlocked` cannot output `null`, so this check isn't needed at all I don't think.

File: src/modules/auth/actions/update-assets.js (hunk 1 above)
Comment (tinybike) [reply src/modules/auth/actions/update-assets.js]: I don't think that eth.getBalance can return null or undefined. Even for a brand new account it should just return 0

File: src/modules/portfolio/actions/collect-market-creator-fees.js (hunk 3 above)
Comment (adrake33) [reply src/modules/portfolio/actions/collect-market-creator-fees.js]: Ok, then the check in `src/modules/auth/actions/update-assets.js` should probably be removed.

File: test/auth/actions/use-unlocked-account-test.js (hunk 5 above)
Comment (adrake33) [reply test/auth/actions/use-unlocked-account-test.js]: Yeah, you're right. Sorry -- there's no need to add this check.
